### PR TITLE
gitlab-ci: enable ubuntu 12.04 builds

### PR DESCRIPTION
**Pull request by chipitsine**
Your great patch is much appreciated. We are considering to apply your patch into the SoftEther VPN main tree.

SoftEther VPN Patch Acceptance Policy:
http://www.softether.org/5-download/src/9.patch

You have two options which are described on the above policy.
Could you please choose either option 1 or 2, and specify it clearly on the reply?

- one

### CODE REVIEW

File: src/Mayaqua/CMakeLists.txt [hunk 1]
```diff
@@ -19,11 +19,24 @@ find_package(ZLIB REQUIRED)
...
+if (HAVE_SYS_AUXV)
+  target_link_libraries(mayaqua cpu_features)
+else (HAVE_SYS_AUXV)
+  add_definitions(-DSKIP_CPU_FEATURES)
+endif (HAVE_SYS_AUXV)
+
+find_library(LIBRT_LIBRARIES rt)
+if(LIBRT_LIBRARIES)
+  target_link_libraries(mayaqua rt)
+endif(LIBRT_LIBRARIES)
```
Comment (davidebeatrici) [inline src/Mayaqua/CMakeLists.txt]: I think that there's no need to specify the condition after `else` and `endif`, as the block is very small.

File: src/Mayaqua/CMakeLists.txt [hunk 2]
```diff
@@ -19,11 +19,24 @@ find_package(ZLIB REQUIRED)
 # In some cases libiconv is not included in libc
 find_library(LIB_ICONV iconv)
 
-add_subdirectory(cpu_features)
+if (HAVE_SYS_AUXV)
```
Comment (davidebeatrici) [inline src/Mayaqua/CMakeLists.txt]: Please keep the same style across the `CMakeLists.txt` files (no space before the parenthesis).

**Comment by chipitsine** [comment]
done

File: src/Mayaqua/CMakeLists.txt [hunk 3]
```diff
@@ -19,11 +19,24 @@ find_package(ZLIB REQUIRED)
...
+endif()
 
 target_include_directories(mayaqua PRIVATE cpu_features/include)
 
-target_link_libraries(mayaqua cpu_features OpenSSL::SSL OpenSSL::Crypto Threads::Threads ZLIB::ZLIB)
+target_link_libraries(mayaqua OpenSSL::SSL OpenSSL::Crypto Threads::Threads ZLIB::ZLIB)
+
+if(HAVE_SYS_AUXV)
+  target_link_libraries(mayaqua cpu_features)
+else
```
Comment (davidebeatrici) [inline src/Mayaqua/CMakeLists.txt]: `else()`

**Comment by chipitsine** [comment]
done

**Pull request merged**